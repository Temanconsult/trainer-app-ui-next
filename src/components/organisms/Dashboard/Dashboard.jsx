import React, { useEffect, useState } from 'react'
import { useQuery } from '@apollo/react-hooks'
import LoaderWrapper from '../../atoms/LoaderWrapper'
import CoursesList from '../../molecules/CoursesList'
import coursesQuery from '../../../graphql/queries/courses'

const Dashboard = () => {
  const [courses, setCourses] = useState([])
  const { data, loading } = useQuery(coursesQuery)

  useEffect(() => {
    if (data) {
      setCourses(data.allCourses)
    }
  }, [data])

  return (
    <LoaderWrapper loading={loading}>
      <CoursesList courses={courses} className="" />
    </LoaderWrapper>
  )
}

export default Dashboard
