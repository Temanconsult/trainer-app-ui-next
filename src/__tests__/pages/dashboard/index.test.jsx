import React from 'react'
import { render } from '@testing-library/react'
import Dashboard from '../../../pages/dashboard'

describe('Pages:Dashboard', () => {
  it('renders the dashboard page', () => {
    const { asFragment } = render(<Dashboard />)
    expect(asFragment()).toMatchSnapshot()
  })
})
