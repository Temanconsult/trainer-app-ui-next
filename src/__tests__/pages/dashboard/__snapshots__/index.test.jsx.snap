// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages:Dashboard renders the dashboard page 1`] = `
<DocumentFragment>
  <p>
    Boilerplate content for dashboard
  </p>
  <p>
    Try navigating to
    <a
      href="/"
    >
      the homepage
    </a>
    and observe the URL changing
  </p>
</DocumentFragment>
`;
